backend/main/src/Group/UseCases/CreateContainerUseCase/CreateContainerUseCase.ts

```diff
@@ -0,0 +1,57 @@
+import { IContainerRepository } from "src/Group/Domain/IContainerRepository";
+import { Container, ContainerId } from "src/Group/Domain/Entities/Container";
+import { IUseCase } from "src/Shared";
+import { Err, Ok, Result } from "result-ts-type";
+import {
+	CreateContainerUseCaseErrorType,
+	ICreateContainerUseCase,
+} from "src/Group/UseCases/CreateContainerUseCase/ICreateContainerUseCase";
+import { IContainerDto, containerDtoMapper } from "src/Group/Dtos/ContainerDto";
+import { Food } from "src/Group/Domain/Entities/Food";
+
+/**
+ * Create a new container. You can call this use case without a containername, and then the new container's name will be the default name.
+ * @throws: When you try to create a new container that already existing, the error will throw an error.
+ */
+export class CreateContainerUseCase
+	implements
+		IUseCase<
+			ICreateContainerUseCase,
+			IContainerDto,
+			CreateContainerUseCaseErrorType
+		>
+{
+	private readonly containerRepository: IContainerRepository;
+
+	constructor(containerRepository: IContainerRepository) {
+		this.containerRepository = containerRepository;
+	}
+
+	public async execute(
+		request: ICreateContainerUseCase,
+	): Promise<Result<IContainerDto, CreateContainerUseCaseErrorType>> {
+		const { name } = request;
+
+		const containerIdOrError = ContainerId.create();
+		// TODO create containername value object
+		const containername = name || "default";
```

Comment:
You can define this knowledge inside the container object. It's optional.
And 'container' and 'name'?

Reply:
I'll make another issue and work on it cuz for me it's easier to keep this branch small and close ealier.